lib/src/booleans/boolean_valid.dart

```diff
@@ -0,0 +1,148 @@
+import 'package:turf/polygon_to_line.dart';
+import 'package:turf/src/booleans/boolean_disjoint.dart';
+import 'package:turf/src/booleans/boolean_point_on_line.dart';
+import 'package:turf/src/meta/extensions.dart';
+
+import '../../helpers.dart';
+import '../line_intersect.dart';
+import 'boolean_crosses.dart';
+
+bool checkRingsClose(List<Position> geom) {
+  return (geom[0].lng == geom[geom.length - 1].lng ||
+      geom[0].lat == geom[geom.length - 1].lat);
+}
+
+bool checkRingsForSpikesPunctures(List<Position> geom) {
+  for (var i = 0; i < geom.length - 1; i++) {
+    var point = Point(coordinates: geom[i]);
+    for (var ii = i + 1; ii < geom.length - 2; ii++) {
+      var seg = [geom[ii], geom[ii + 1]];
+      if (booleanPointOnLine(point, LineString(coordinates: seg))) return true;
+    }
+  }
+  return false;
+}
+
+bool checkPolygonAgainstOthers(Polygon poly, MultiPolygon geom, int index) {
```

Comment:
shoudn't these functions be private with`_`, or should they be exposed to other files / the library user?